### PR TITLE
Test metric

### PR DESCRIPTION
**Pull request by ahn1340**

### CODE REVIEW

File: test/test_metric/test_metrics.py [hunk 1]
```diff
@@ -308,3 +308,137 @@ def test_classification_metrics(self):
...
+class TestMetric(unittest.TestCase):
+
+    def test_regression_all(self):
+
+        for metric, scorer in autosklearn.metrics.REGRESSION_METRICS.items():
+            y_true = np.array([1, 2, 3, 4])
+            y_pred = y_true.copy()
+
+            # the best possible score of r2 loss is 1.
+            if metric == 'r2':
```
Comment (mfeurer) [inline test/test_metric/test_metrics.py]: Could you use the optimum attribute here?

File: test/test_metric/test_metrics.py [hunk 2]
```diff
@@ -308,3 +308,137 @@ def test_classification_metrics(self):
...
+            current_score = scorer(y_true, y_pred)
+            self.assertLess(current_score, previous_score)
+
+
+    def test_classification_binary(self):
+
+        for metric, scorer in autosklearn.metrics.CLASSIFICATION_METRICS.items():
+            # Skip functions not applicable for binary classification.
+            # TODO: Average precision should work for binary classification,
+            # TODO: but its behavior is not right.
```
Comment (mfeurer) [inline test/test_metric/test_metrics.py]: not right in what sense?

File: test/test_metric/test_metrics.py [hunk 3]
```diff
@@ -308,3 +308,137 @@ def test_classification_metrics(self):
...
+            if metric in ['average_precision', 'pac_score',
+                          'precision_samples', 'recall_samples', 'f1_samples']:
+                continue
+
+            y_true = np.array([1.0, 1.0, 1.0, 0.0, 0.0, 0.0])
+            y_pred = np.array([[0.0, 1.0], [0.0, 1.0], [0.0, 1.0], [1.0, 0.0],
+                                [1.0, 0.0], [1.0, 0.0]])
+            if metric is 'log_loss':
```
Comment (mfeurer) [inline test/test_metric/test_metrics.py]: Could this be done via the `optimum` attribute of the scorer?

File: test/test_metric/test_metrics.py [hunk 4]
```diff
@@ -308,3 +308,137 @@ def test_classification_metrics(self):
...
+            # Skip functions not applicable for multiclass classification.
+            if metric in ['pac_score', 'roc_auc', 'average_precision',
+                          'precision', 'recall', 'f1','precision_samples',
+                          'recall_samples', 'f1_samples']:
+                continue
+            y_true = np.array([0.0, 0.0, 1.0, 1.0, 2.0])
+            y_pred = np.array([[1.0, 0.0, 0.0], [1.0, 0.0, 0.0],
+                            [0.0, 1.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]])
+            if metric is 'log_loss': # the best possible score for log_loss is 0.
+                previous_score = 0
```
Comment (mfeurer) [inline test/test_metric/test_metrics.py]: Could this be done via the `optimum` attribute of the scorer?

File: test/test_metric/test_metrics.py [hunk 5]
```diff
@@ -308,3 +308,137 @@ def test_classification_metrics(self):
...
+    def test_classification_multilabel(self):
+
+        for metric, scorer in autosklearn.metrics.CLASSIFICATION_METRICS.items():
+            # Skip functions not applicable for multi-label classification.
+            if metric in ['roc_auc', 'log_loss',
+                          'pac_score', 'precision', 'recall', 'f1']:
+                continue
+            y_true = np.array([[1, 0, 0], [1, 1, 0], [0, 1, 1], [1, 1, 1]])
+            y_pred = y_true.copy()
+            previous_score = 1
```
Comment (mfeurer) [inline test/test_metric/test_metrics.py]: Could this be done via the `optimum` attribute of the scorer?
